spark/src/main/scala/io/substrait/spark/logical/ToSubstraitRel.scala

```diff
@@ -72,7 +71,8 @@ class ToSubstraitRel extends AbstractLogicalPlanVisitor with Logging {
     val substraitExps = expression.aggregateFunction.children.map(toExpression(output))
     val invocation =
       SparkExtension.toAggregateFunction.apply(expression, substraitExps)
-    relation.Aggregate.Measure.builder.function(invocation).build()
+    val filter = expression.filter map toExpression(output)
+    relation.Aggregate.Measure.builder.function(invocation).preMeasureFilter(Optional.ofNullable(filter.orNull)).build()
```

Comment:
I assume including the filter is an opportunistic bug fix?

I don't think it needs to be it's own PR, but in the future if you could make it a separate commit with a `fix(spark)` style message it makes it easier for me to notice and call out such changes in the release log.

Reply:
Sure, I've split this into a separate commit.